### DISCUSSION TITLE
Periodic authentication errors when retrieving records

### DISCUSSION THREAD

**Discussion by clwdesign** [AlexaCRM/dynamics-webapi-toolkit#74] [Q&A]
Periodically, I will start to get the following error when doing a simple Retrieve request -- e.g., $client->Retrieve('contact', 'SOMEIDHERE', new \AlexaCRM\Xrm\ColumnSet( true ));

"Authentication at Azure AD failed. cURL error 6: Could not resolve host: login.microsoftonline.com (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)"

I am not changing anything in my request and the exact same request had been successful one minute before and will be successful one minute after. Am I hitting some kind of intermittent usage limitation?

I am making maybe 10 requests in a 5-minute period so pretty tiny.

**Comment by georged** [comment]
@clwdesign it really sounds like intermittent DNS issues on your hosting server.

**Reply by clwdesign** [reply]
Thanks very much for replying. I am making the requests in a local testing environment built with Docker. I will try from other environments.

**Reply by georged** [reply]
Docker 👀 - that'd be my first suspect.

**Reply by clwdesign** [reply]
Great, thank you for pointing me in this direction. I will investigate.
